### PR TITLE
Fix minor typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Just thought I'd contribute some typo fixes I stumbled upon. Nothing controversial (hopefully), just 74 simple fixes.

Use the following command to get a quick and dirty summary of the specific corrections made:
```shell
git diff HEAD^! --word-diff-regex='\w+' -U0 \
  | grep -E '\[\-.*\-\]\{\+.*\+\}' \
  | sed -r 's/.*\[\-(.*)\-\]\{\+(.*)\+\}.*/\1 \2/' \
  | sort | uniq -c | sort -n
```

FWIW, the top typos are:
* satisifed (8)
* uncommited (7)
* tranparent (7)
* expecially (3)
* recieves (3)
* correponding (2)
* slighly (2)
* wierdly (2)

### CODE REVIEW

**Review by eugeneo — COMMENTED** [review]
Thank you for your contributions. This looks great, I will approve the PR but we will need to run a lot of tests as it touches things like env variable names.

**Comment by NathanBaulch** [comment]
Are you referring to `STABLE_GRPC_UNCOMMITED_PATCH_CHECKSUM` under `/tools/bazelify_tests` perhaps? This is closest thing to an env var that I touched, happy to revert that change if too risky.

**Comment by eugeneo** [comment]
Please note there's relevant CI failure:
```
< # Use the FORMULA_TRANSPARENT tag to determine whether or not the images
---
> # Use the FORMULA_TRANPARENT tag to determine whether or not the images
Traceback (most recent call last):
  File "tools/buildgen/generate_projects.py", [line 154](https://cs.corp.google.com/piper///depot/google3/tools/buildgen/generate_projects.py?l=154&ws&snapshot=0), in <module>
    main()
  File "tools/buildgen/generate_projects.py", [line 146](https://cs.corp.google.com/piper///depot/google3/tools/buildgen/generate_projects.py?l=146&ws&snapshot=0), in main
    assert 0 == os.system("diff %s %s" % (s, g)), s
AssertionError: ./tools/doxygen/Doxyfile.php
```
Make sure you run `./tools/buildgen/generate_projects.sh` before pushing the update.

**Comment by eugeneo** [comment]
I believe other failure are unrelated flakes.

**Comment by NathanBaulch** [comment]
I'm not familiar with Doxygen but it looks like that typo was fixed 7 years ago in [v1.8.14](https://github.com/doxygen/doxygen/commit/f88ebaf9f23c3151f312400d77150eeeeb8158c2) which is newer than the version stamp in the 12 Doxyfile files in this repo (v1.8.5 - v1.8.13).
Nevermind, I've gone ahead and restored that typo.

**Comment by eugeneo** [comment]
I created a PR in your repo: https://github.com/NathanBaulch/grpc/pull/1

**Comment by NathanBaulch** [comment]
I think I found a better fix @eugeneo - update the doxygen and dockerfile templates instead! Hopefully [this](https://github.com/grpc/grpc/pull/37450/commits/44553539e6ec545205ad8ef2f7505fb44152bb33#diff-3eb5b818229b040d7a6ac024995ab48b0df7fc44a01856352ad6d13f0f1f48f9) change solves the project gen issue.

**Comment by NathanBaulch** [comment]
So sorry @eugeneo, I missed one more doxygen template file! Can I have `kokoro:run` again? 🤞 🙏

File: tools/dockerfile/test/cxx_gcc_7_x64/Dockerfile [hunk 1]
```diff
@@ -58,7 +58,7 @@ RUN python3 -m pip install twisted
 RUN apt-get update && apt-get install -y cmake && apt-get clean
 
 #=================
-# Install ccache (use slighly older release of ccache that works older compilers and cmake)
```
Comment (eugeneo) [inline tools/dockerfile/test/cxx_gcc_7_x64/Dockerfile:61]: I think this change changes the MD5 hash so CI get's worried the image needs to be rebuilt... Please revert this change.

File: tools/dockerfile/test/cxx_gcc_8_x64/Dockerfile [hunk 2]
```diff
@@ -58,7 +58,7 @@ RUN python3 -m pip install twisted
 RUN apt-get update && apt-get install -y cmake && apt-get clean
 
 #=================
-# Install ccache (use slighly older release of ccache that works older compilers and cmake)
+# Install ccache (use slightly older release of ccache that works older compilers and cmake)
```
Comment (eugeneo) [inline tools/dockerfile/test/cxx_gcc_8_x64/Dockerfile:61]: Please revert this change

**Review by eugeneo — COMMENTED** [review]
So sorry for the pain. This change is really nice but it triggers some CI checks fixing which can take quite some time.

**Comment by NathanBaulch** [comment]
No worries at all @eugeneo, a PR that touches so many unrelated files on a mature well tested project like this is always going to have hiccups. I have another batch of fixes too but we'll get through this batch first.

**Comment by eugeneo** [comment]
Looks like flaky tests, I will try rerunning them.

**Review by eugeneo — APPROVED** [review]
Thanks for the contribution! I am looking into what's up with the CI...

**Review by Vignesh2208 — APPROVED** [review]
